application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -226,6 +240,31 @@ private ListenableFuture<TransportApiResponseMsg> validateCredentials(TransportP
         }
     }
 
+    private ListenableFuture<TransportApiResponseMsg> validateDeviceProfileCertificate(String credentialsId) {
+        DeviceProfile deviceProfile = deviceProfileService.findDeviceProfileByCertificateHash(credentialsId);
+        if (deviceProfile != null) {
+            return getDeviceProfileInfo(deviceProfile);
+        }
+        return getEmptyTransportApiResponseFuture();
+    }
+
+    private ListenableFuture<TransportApiResponseMsg> updateOrCreateDeviceCredentials(String credentialsId,
+                                                                                      String credentialsValue,
+                                                                                      String deviceCN,
+                                                                                      DeviceProfile deviceProfile,
+                                                                                      DeviceCredentialsType credentialsType) {
+        String deviceName = extractRegex(deviceCN, deviceProfile.getCertificateRegexPattern());
+        // find deviceCredentials by deviceName (device exists)
+        DeviceCredentials deviceCredentials = deviceCredentialsService.findDeviceCredentialsByTenantIdAndDeviceName(deviceProfile.getTenantId(), deviceName);
```

Comment:
I would reuse **deviceService.findDeviceByName** and do not create this method. 
What are the benefits of this new method findDeviceCredentialsByTenantIdAndDeviceName?